src/mame/video/tecmo_mix.cpp

```diff
@@ -89,6 +89,28 @@ void tecmo_mix_device::set_bgpen(device_t &device, int bgpen)
 	dev.m_bgpen = bgpen;
 }
 
+uint32_t tecmo_mix_device::sum_colors (const pen_t *pal, int c1_idx, int c2_idx)
+{
+	pen_t c1 = pal [c1_idx];
+	pen_t c2 = pal [c2_idx];
+	
+	uint8_t c1_a = c1 >> 24;
+	uint8_t c1_r = c1 >> 16;
+	uint8_t c1_g = c1 >> 8;
+	uint8_t c1_b = c1;
+
+	uint8_t c2_a = c2 >> 24;
+	uint8_t c2_r = c2 >> 16;
+	uint8_t c2_g = c2 >> 8;
+	uint8_t c2_b = c2;
+
+	uint8_t a = std::min (0xFF, c1_a + c2_a);
+	uint8_t r = std::min (0xFF, c1_r + c2_r);
+	uint8_t g = std::min (0xFF, c1_g + c2_g);
+	uint8_t b = std::min (0xFF, c1_b + c2_b);
+
+	return ((a << 24) | (r << 16) | (g << 8) | b);
+}
```

Comment:
Could you please change the following:
* Don't use gnu style, we prefer Allaman or K&R (i.e. use `func(arg)` and `arr[idx]` rather than `func (arg)` and `arr [idx]`)
* Make all your intermediates `const` if you're not planning to change them
* Don't play fast and loose with integers - you have all kinds of problems with overflow and abuse of `std::min`

The code definitely doesn't work as intended, the `std::min` calls are effectively doing nothing.  The second argument is the sum of two `uint8_t` which will not be promoted, so they'll exhibit modulo maths and the result of the addition will always be less than `0xff`.  You need to either make the intermediates larger (making them `unsigned int` is sensible, as it's going to be a type that's efficient to use), or explicitly cast one or both to something larger when adding.

Also, `0xff` is an `int` literal while the sum of two `uint8_t` is a `uint8_t` so the overload of `std::min` is ambiguous.